liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/CreateServerMojo.java

```diff
@@ -174,4 +175,39 @@ private File getFileFromConfigDirectory(String file, File def) {
         } 
         return null;
     }
+    
+    /*
+     * return the filename of the project artifact to be installed by install-apps goal
+     */
+    private String getApplicationFilename() {
+        // project artifact has not be created yet when create-server goal is called in pre-package phase
+        String name = project.getBuild().getFinalName();
+        if (stripVersion) {
+            int versionBeginIndex = project.getBuild().getFinalName().lastIndexOf("-" + project.getVersion());
+            if ( versionBeginIndex != -1) {
+                name = project.getBuild().getFinalName().substring(0, versionBeginIndex);
+            }
+        }
+        
```

Comment:
Did you also notice that, I am checking valid application types, ear, web, eba and esa as documented in our Liberty knowledge center. https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/twlp_dep.html.  This check should also go into installAppMojo.  I will make that change under the loose config issue.